idaes/power_generation/carbon_capture/compression_system/compressor.py

```diff
@@ -0,0 +1,659 @@
+Control of a Multistage CO2 Compresor System." International Journal
+of Greenhouse Gas Control. v62., page 31-45
+Created: April 2020
+__Author__ = "Quang Minh Le, John Eslick, Andrew Lee"
+"""
+
+from __future__ import division
+from pyomo.environ import (SolverFactory,
+                           NonNegativeReals, Var, Param,
+                           value, log, sqrt, exp,
+                           units as pyunits)
+from pyomo.common.config import ConfigValue, In
+from idaes.core import declare_process_block_class
+
+from idaes.generic_models.unit_models.pressure_changer import (
+    PressureChangerData, ThermodynamicAssumption)
+
+import idaes.logger as idaeslog
+from idaes.core.util.exceptions import ConfigurationError
+from idaes.core.util.constants import Constants as const
+from enum import Enum
+
+_log = idaeslog.getLogger(__name__)
+
+
+class ImpellerType(Enum):
+    cover_impeller = 1
+    open_impeller = 2
+    custom = 3
+
+
+def _cover_impeller_callback(blk):
+    blk.mass_flow_coeff = Var(blk.flowsheet().config.time,
+                              initialize=0.0735,
+                              doc="Compressor Flow Coefficient",
+                              bounds=(0.01, 0.15))
+
+    @blk.Constraint(blk.flowsheet().config.time)
+    def impeller_work_coeff_eqn(b, t):
+        phi = blk.mass_flow_coeff[t]
+        return b.impeller_work_coeff[t] * 100 * phi == (
+                          0.62 * phi - phi * (phi / 0.4)**3 + 0.0014) * 100
+
+    @blk.Constraint(blk.flowsheet().config.time)
+    def polytropic_head_coeff_vaned_diffuser_eqn(b, t):
+        phi = blk.mass_flow_coeff[t]
+        return b.mu_p_v[t] * 100 * phi == (
+                      0.51 * phi + phi**2 - 7.6 * phi**3 - 0.00025) * 100
+
+
+def _open_impeller_callback(blk):
+    blk.mass_flow_coeff = Var(blk.flowsheet().config.time,
+                              initialize=0.0735,
+                              doc="Compressor Flow Coefficient",
+                              bounds=(0.01, 0.15))
+
+    @blk.Constraint(blk.flowsheet().config.time)
+    def impeller_work_coeff_eqn(b, t):
+        phi = blk.mass_flow_coeff[t]
+        return b.impeller_work_coeff[t] * phi == (
+                      0.68 * phi - phi * (phi / 0.37)**3 + 0.002)
+
+    @blk.Constraint(blk.flowsheet().config.time)
+    def polytropic_head_coeff_vaned_diffuser_eqn(b, t):
+        phi = blk.mass_flow_coeff[t]
+        return b.mu_p_v[t] * phi == 0.5 * (1 + (phi - 0.065) / abs(
+            phi - 0.065)) * (
+                0.59 * phi + 0.7 * phi**2 - 7.5 * phi**3 - 0.00025
+                ) + 0.5 * (1 - (phi - 0.065) / abs(phi - 0.065)) * 0.6 * phi
+
+
+class VaneDiffuserType(Enum):
+    vane_diffuser = 1
+    vaneless_diffuser = 2
+    custom = 3
+
+
+def _vane_diffuser_callback(blk):
+    @blk.Constraint(blk.flowsheet().config.time)
+    def polytropic_head_coeff_eqn(b, t):
+        return b.mu_p[t] == b.mu_p_v[t]
+
+    def polytropic_efficiency_eqn(b, t):
```

Comment:
It looks like you are missing a decorator here.

Reply:
Hi @eslickj , @andrewlee94 : I got the same error when running a test. Can you check the latest one? Thanks!
TypeError: unsupported operand type(s) for *: 'int' and 'dict'

Reply:
Hi @eslickj @andrewlee94 : I uploaded the latest model. I still cannot get the convergence due to the below error. Please help me check the code again. Thanks!
  TypeError: unsupported operand type(s) for *: 'int' and 'dict'